### ISSUE TITLE
Respond to Dan's Feedback And Questions

### ISSUE THREAD

**Issue by danielhalm1407** [danielhalm1407/st310_project#20]
![Image](https://github.com/user-attachments/assets/8f6868f3-0187-4d49-821d-749426585078)

**Comment by danielhalm1407** [comment]
Finish this and add some domain research as well.

**Comment by sintayehuan** [comment]
Here is my progress as of now:

- [x] Removed the `split_data` method in favor of the tidymodel `training()` and `testing()` methods
- [x] Implemented the Barzilai-Borwein step size method to the `gradient_descent` method
- [x] Implemented the first out of two visualizations (predictors versus coefficients across our model implementation, glm, and our model on simulated data "naive model")
- [x] Implemented the second out of two visualizations (time series of `betas` over each step/iteration)

Here is what I am currently working on:

- [x] Running the algorithm on simulated data** to see "naive accuracy" later
- [x] Implementing a heatmap confusion matrix (is this for a visualization in the accuracy section?)
- [ ] Plot the ROC curve and explain the AUC/ROC section
- [ ] Explain the BB algorithm and both new visualizations

\** Daniel, how are you simulating the data? Should I just using `rnorm`, `rbinom`, etc. and make a similar looking dataset to run our implementation of gradient descent on?

I'll finish what I am currently working on and then all I have left to do is more explanation (answering Daniel's questions) and commenting. I should have a lot (if not all) done by the time we meet on Tuesday.

**Comment by danielhalm1407** [comment]
** Daniel, how are you simulating the data? Should I just using rnorm, rbinom, etc. and make a similar looking dataset to run our implementation of gradient descent on?

Doesn't have to be similar to our data. In fact, better if it isn't. You can use the code from the seminars for generating these, which happens to use runif, but any distribution works. I would use runif just for the sake of argument. I've put it here below:


# Setting parameters
set.seed(1)
# Try changing some parameters
n <- 200
p <- 10
sparsity <- p/2 # only half of our indep vars are relevant
nonzero_beta <- runif(sparsity, min = -1, max = 1)
# or, e.g. rep(1, sparsity), runif(sparsity, min = -1, max = 1)
true_beta <- c(.5, nonzero_beta, rep(0, p - sparsity))

**Comment by danielhalm1407** [comment]
- Finish the write-up
- Citations and bibliography
- Add the lasso penalty to the loss function of the logit in gradient decsnet
- copy and paste the stochastic gd and see if it just runs nicely. if it takes more than 1 hour, just drop
